### ISSUE TITLE
Run integration tests on PR

### ISSUE THREAD

**Issue by cka-y** [MobilityData/mobility-feed-api#549]
### Describe the problem

Currently, we have to wait until a PR is merged into the main branch before running the integration tests in the QA environment. This approach has several drawbacks:

- **Delayed Feedback:** We often encounter failing integration tests only after multiple PRs have been merged, which delays the identification and resolution of issues.
- **Complex Debugging:** Identifying the root cause of a failure becomes more complicated when several PRs have been merged, making it harder to pinpoint the exact source of the bug.

### Proposed solution

Modify the integration tests script to allow it to run locally. Specifically, configure it to execute as part of the GitHub Actions workflow when a PR is created or updated. This can be achieved by either tunneling to the QA database or populating a local database for testing purposes.
